packages/wonder-blocks-data/util/response-cache.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import type {
+    CacheEntry,
+    ResponseCache as Cache,
+    IRequestHandler,
+} from "./types.js";
+
+function deepClone<T: {...}>(source: T | $ReadOnly<T>): $ReadOnly<T> {
+    /**
+     * We want to deep clone the source cache to dodge mutations by external
+     * references. So we serialize the source cache to JSON and parse it
+     * back into a new object.
+     */
+    const serializedInitCache = JSON.stringify(source);
+    const cloneInitCache = JSON.parse(serializedInitCache);
+    return Object.freeze(cloneInitCache);
+}
+
+/**
+ * The default instance is stored here.
+ * It's created below in the Default() static property.
+ */
+let _default: ResponseCache;
+
+/**
+ * Implements the response cache.
+ *
+ * INTERNAL USE ONLY
+ */
+export class ResponseCache {
+    static get Default() {
+        if (!_default) {
+            _default = new ResponseCache();
+        }
+        return _default;
+    }
+
+    _cache: Cache = {};
+
+    constructor(source?: ?Cache = undefined) {
```

Comment:
I think this can be `source: ?Cache = undefined` since default params imply optionality.

Reply:
I tried that and it gives an error.